tests/object_explorer/test_object_explorer_service.py

```diff
@@ -87,31 +92,33 @@ def test_register(self):
     def test_generate_uri_missing_params(self):
         # Setup: Create the parameter sets that will be missing a param each
         params = [
-            ConnectionDetails.from_data({'host': None, 'dbname': TEST_DBNAME, 'user': TEST_USER}),
-            ConnectionDetails.from_data({'host': TEST_HOST, 'dbname': None, 'user': TEST_USER}),
-            ConnectionDetails.from_data({'host': TEST_HOST, 'dbname': TEST_DBNAME, 'user': None})
+            ConnectionDetails.from_data({'host': None, 'dbname': TEST_DBNAME, 'user': TEST_USER, 'port': TEST_PORT}),
+            ConnectionDetails.from_data({'host': TEST_HOST, 'dbname': None, 'user': TEST_USER, 'port': TEST_PORT}),
+            ConnectionDetails.from_data({'host': TEST_HOST, 'dbname': TEST_DBNAME, 'user': None, 'port': TEST_PORT}),
+            ConnectionDetails.from_data({'host': TEST_HOST, 'dbname': TEST_DBNAME, 'user': TEST_USER, 'port': None})
         ]
 
         for param_set in params:
             # If: I generate a session URI from params that are missing a value
             # Then: I should get an exception
             with self.assertRaises(Exception):
-                ObjectExplorerService._generate_session_uri(param_set)
+                ObjectExplorerService._generate_session_uri(param_set, PG_PROVIDER_NAME)
 
     def test_generate_uri_valid_params(self):
         # If: I generate a session URI from a valid connection details object
         params, session_uri = _connection_details()
-        output = ObjectExplorerService._generate_session_uri(params)
+        output = ObjectExplorerService._generate_session_uri(params, PG_PROVIDER_NAME)
 
         # Then: The output should be a properly formed URI
         parse_result = url_parse.urlparse(output)
         self.assertEqual(parse_result.scheme, 'objectexplorer')
         self.assertTrue(parse_result.netloc)
 
-        re_match = re.match(r'(?P<username>\w+)@(?P<host>\w+):(?P<db_name>\w+)', parse_result.netloc)
+        re_match = re.match(r'(?P<username>\w+)@(?P<host>\w+):(?P<port>\w+):(?P<db_name>\w+)', parse_result.netloc)
```

Comment:
Why do we need to change this regex?

Reply:
The session uri now includes a port number, so the regex needs to be changed to account for that, so that it correctly grabs the correct information for the assertions below.

Reply:
Oh, did it change in the product code?